modules/system/defaults/screencapture.nix

```diff
@@ -0,0 +1,16 @@
+{ config, lib, ... }:
+
+with lib;
+
+{
+  options = {
+
+    system.defaults.screencapture.location = mkOption {
+      type = types.string;
+      default = "~/Desktop";
```

Comment:
Defaults usually have a null value, in which case it's just ignored.

Reply:
So I did initially use `null` here, but figured a default of `"~/Desktop"` was better.  
Reason being; if it's `null`, when a user decides they want to stop customising this property, and they comment out/remove their configuration for it, it will just remain the same as they had it before.  

Whereas, if it defaults to `"~/Desktop"`, when removed, it'll switch back to that - which is the default the OS comes with.  

Of course, it's up for debate if a user should expect this behaviour  - I just felt it was more natural, but happy to go with what you think is best.

Reply:
The problem is that we don't know if it still has the default value. I'm not sure how to handle this properly yet, but I don't want to somebody's settings to all get reset because of running nix-darwin with an empty config.

Reply:
Sure, makes sense - I agree with that. I'll set a default of `null` then

Reply:
I don't really like the null values either, it makes rollback/removing values have unexpected behaviour from nix-darwin's perspective. Something like an opt-in 'strict' mode for defaults could be an easy solution for that.